src/stm32f1/spi.c

```diff
@@ -97,9 +104,10 @@ spi_transfer(struct spi_config config, uint8_t receive_data,
     while (len--) {
         LL_SPI_TransmitData8(SPI2, *data);
         while (!LL_SPI_IsActiveFlag_TXE(SPI2));
+        while (!LL_SPI_IsActiveFlag_RXNE(SPI2));
+        uint8_t rdata = LL_SPI_ReceiveData8(SPI2);
```

Comment:
Always read the incoming SPI data, do nothing with it if a read was not requested.